secret keyring panics, "no annotations"

## Describe the bug

Any attempt to use the keyring results in a panic "no annotations".

## To reproduce

Attempt to run `chezmoi secret keyring set` or `chezmoi secret keyring get`.

## Expected behavior

Adding or retrieving a value from the keyring.

## Output of command with the `--verbose` flag

```console
chezmoi --verbose secret keyring get --service=cz --user=test
panic: "get": no annotations

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.getAnnotations(0x140001f1200)
	github.com/twpayne/chezmoi/v2/internal/cmd/annotation.go:47 +0x130
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0x140003a8000, 0x140001f1200, {0x140007279c8?, 0x100aa9e4c?, 0x14000284000?})
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:1886 +0x30
github.com/spf13/cobra.(*Command).execute(0x140001f1200, {0x1400027c090, 0x3, 0x3})
	github.com/spf13/cobra@v1.8.0/command.go:954 +0x724
github.com/spf13/cobra.(*Command).ExecuteC(0x14000327800)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0x14000727d08?, {0x14000150010, 0x6, 0x6})
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:1216 +0x70
github.com/twpayne/chezmoi/v2/internal/cmd.runMain({{0x101ad75dc, 0x6}, {0x101ad8800, 0x28}, {0x101ad8580, 0x14}, {0x101ad8490, 0x8}}, {0x14000150010, 0x6, ...})
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:298 +0x180
github.com/twpayne/chezmoi/v2/internal/cmd.Main({{0x101ad75dc, 0x6}, {0x101ad8800, 0x28}, {0x101ad8580, 0x14}, {0x101ad8490, 0x8}}, {0x14000150010?, 0x6, ...})
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:111 +0x60
main.main()
	github.com/twpayne/chezmoi/v2/main.go:26 +0xd8
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
RESULT   CHECK                       MESSAGE
ok       version                     v2.42.3, commit 43a4f5790ef1b709de881d36afb0a61745efa34f, built at 2023-12-16T23:08:59Z, built by Homebrew
ok       latest-version              v2.42.3
ok       os-arch                     darwin/arm64
ok       uname                       Darwin m.local 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000 arm64
ok       go-version                  go1.21.5 (gc)
ok       executable                  /opt/homebrew/bin/chezmoi
ok       upgrade-method              brew-upgrade
ok       config-file                 ~/.config/chezmoi/chezmoi.toml, last modified 2023-12-20T12:49:42-06:00
ok       source-dir                  ~/.local/share/chezmoi is a git working tree (clean)
ok       suspicious-entries          no suspicious entries
ok       working-tree                ~/.local/share/chezmoi is a git working tree (clean)
ok       dest-dir                    ~ is a directory
ok       umask                       022
ok       cd-command                  found /bin/zsh
ok       cd-args                     /bin/zsh
info     diff-command                not set
ok       edit-command                found /usr/bin/vi
ok       edit-args                   /usr/bin/vi
ok       git-command                 found /opt/homebrew/bin/git, version 2.43.0
ok       merge-command               found /usr/bin/vimdiff
ok       shell-command               found /bin/zsh
ok       shell-args                  /bin/zsh
info     age-command                 age not found in $PATH
ok       gpg-command                 found /opt/homebrew/bin/gpg, version 2.4.3
info     pinentry-command            not set
info     1password-command           op not found in $PATH
info     bitwarden-command           bw not found in $PATH
info     bitwarden-secrets-command   bws not found in $PATH
info     dashlane-command            dcli not found in $PATH
info     doppler-command             doppler not found in $PATH
info     gopass-command              gopass not found in $PATH
info     keepassxc-command           keepassxc-cli not found in $PATH
info     keepassxc-db                not set
info     keeper-command              keeper not found in $PATH
info     lastpass-command            lpass not found in $PATH
info     pass-command                pass not found in $PATH
info     passhole-command            ph not found in $PATH
info     rbw-command                 rbw not found in $PATH
info     vault-command               vault not found in $PATH
info     vlt-command                 vlt not found in $PATH
info     secret-command              not set
```

</details>

## Additional context

Add any other context about the problem here.

--- a/internal/cmd/agecmd.go
+++ b/internal/cmd/agecmd.go
@@ -25,10 +25,9 @@ type ageCmdConfig struct {
 
 func (c *Config) newAgeCmd() *cobra.Command {
 	ageCmd := &cobra.Command{
-		Use:         "age",
-		Args:        cobra.NoArgs,
-		Short:       "Interact with age",
-		Annotations: newAnnotations(),
+		Use:   "age",
+		Args:  cobra.NoArgs,
+		Short: "Interact with age",
 	}
 
 	ageDecryptCmd := &cobra.Command{

--- a/internal/cmd/helpcmd.go
+++ b/internal/cmd/helpcmd.go
@@ -9,12 +9,14 @@ import (
 
 func (c *Config) newHelpCmd() *cobra.Command {
 	helpCmd := &cobra.Command{
-		Use:         "help [command]",
-		Short:       "Print help about a command",
-		Long:        mustLongHelp("help"),
-		Example:     example("help"),
-		RunE:        c.runHelpCmd,
-		Annotations: newAnnotations(),
+		Use:     "help [command]",
+		Short:   "Print help about a command",
+		Long:    mustLongHelp("help"),
+		Example: example("help"),
+		RunE:    c.runHelpCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 
 	return helpCmd

--- a/internal/cmd/internaltestcmd.go
+++ b/internal/cmd/internaltestcmd.go
@@ -21,6 +21,9 @@ func (c *Config) newInternalTestCmd() *cobra.Command {
 		Args:  cobra.MinimumNArgs(1),
 		Short: "Run promptBool",
 		RunE:  c.runInternalTestPromptBoolCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	internalTestCmd.AddCommand(internalTestPromptBoolCmd)
 
@@ -29,6 +32,9 @@ func (c *Config) newInternalTestCmd() *cobra.Command {
 		Args:  cobra.MinimumNArgs(2),
 		Short: "Run promptChoice",
 		RunE:  c.runInternalTestPromptChoiceCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	internalTestCmd.AddCommand(internalTestPromptChoiceCmd)
 
@@ -37,6 +43,9 @@ func (c *Config) newInternalTestCmd() *cobra.Command {
 		Args:  cobra.MinimumNArgs(1),
 		Short: "Run promptInt",
 		RunE:  c.runInternalTestPromptIntCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	internalTestCmd.AddCommand(internalTestPromptIntCmd)
 
@@ -45,6 +54,9 @@ func (c *Config) newInternalTestCmd() *cobra.Command {
 		Args:  cobra.MinimumNArgs(1),
 		Short: "Run promptString",
 		RunE:  c.runInternalTestPromptStringCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	internalTestCmd.AddCommand(internalTestPromptStringCmd)
 
@@ -53,6 +65,9 @@ func (c *Config) newInternalTestCmd() *cobra.Command {
 		Args:  cobra.NoArgs,
 		Short: "Read a password",
 		RunE:  c.runInternalTestReadPasswordCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	internalTestCmd.AddCommand(internalTestReadPasswordCmd)
 

--- a/internal/cmd/licensecmd.go
+++ b/internal/cmd/licensecmd.go
@@ -11,13 +11,15 @@ import (
 
 func (c *Config) newLicenseCmd() *cobra.Command {
 	licenseCmd := &cobra.Command{
-		Use:         "license",
-		Short:       "Print license",
-		Long:        mustLongHelp("license"),
-		Example:     example("license"),
-		Args:        cobra.NoArgs,
-		RunE:        c.runLicenseCmd,
-		Annotations: newAnnotations(),
+		Use:     "license",
+		Short:   "Print license",
+		Long:    mustLongHelp("license"),
+		Example: example("license"),
+		Args:    cobra.NoArgs,
+		RunE:    c.runLicenseCmd,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 
 	return licenseCmd

--- a/internal/cmd/secretkeyringcmd.go
+++ b/internal/cmd/secretkeyringcmd.go
@@ -41,6 +41,9 @@ func (c *Config) newSecretKeyringCmd() *cobra.Command {
 		Args:  cobra.NoArgs,
 		Short: "Delete a value from keyring",
 		RunE:  c.runSecretKeyringDeleteCmdE,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	secretKeyringDeletePersistentFlags := keyringDeleteCmd.PersistentFlags()
 	secretKeyringDeletePersistentFlags.StringVar(
@@ -58,6 +61,9 @@ func (c *Config) newSecretKeyringCmd() *cobra.Command {
 		Args:  cobra.NoArgs,
 		Short: "Get a value from keyring",
 		RunE:  c.runSecretKeyringGetCmdE,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	secretKeyringGetPersistentFlags := keyringGetCmd.PersistentFlags()
 	secretKeyringGetPersistentFlags.StringVar(
@@ -75,6 +81,9 @@ func (c *Config) newSecretKeyringCmd() *cobra.Command {
 		Args:  cobra.NoArgs,
 		Short: "Set a value in keyring",
 		RunE:  c.runSecretKeyringSetCmdE,
+		Annotations: newAnnotations(
+			doesNotRequireValidConfig,
+		),
 	}
 	secretKeyringSetPersistentFlags := keyringSetCmd.PersistentFlags()
 	secretKeyringSetPersistentFlags.StringVar(

